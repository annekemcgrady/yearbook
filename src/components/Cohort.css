html {
  height: 100%;
  overflow: hidden;
}

.card-area {
display: flex;
flex-wrap: wrap;
justify-content: center;
}

h2 {
  background-color: rgba(255, 255, 255, 0.95);
  padding: 10px;
  color: #6495EC;
}
