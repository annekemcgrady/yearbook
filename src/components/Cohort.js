import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = props => {
  const peopleCards = props.data.map(person => {
    return <Person id={person.id} photo={person.photo || 'https://placekitten.com/200/300'} name={person.name} quote={person.quote} superlative={person.superlative} />;
  });
  return (
    <div>
      <h2>{props.title}</h2>
      <div className="card-area">
      {peopleCards}
      </div>
    </div>
  )
}

export default Cohort;
