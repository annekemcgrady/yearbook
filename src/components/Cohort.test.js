import React from 'react';
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort.js';

describe('Cohort', () => {

 const fakeData = [{id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'},
  {id: 2, name: 'Amy', quote: 'I believe in you, cuties!', superlative: 'Most Likely to Be an Actual Superhero', photo: './amy.jpg'}]
  
  it('should match the snapshot when person data is passed in correctly', () => { 
    const wrapper = mount( 
    <Cohort 
      title="Staff"
      data = {fakeData} 
    />);

    expect(wrapper).toMatchSnapshot()
  })
})