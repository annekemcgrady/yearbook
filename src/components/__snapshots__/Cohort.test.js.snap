// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot when person data is passed in correctly 1`] = `
<Cohort
  data={
    Array [
      Object {
        "id": 1,
        "name": "Eric",
        "photo": "./eric.jpg",
        "quote": "You miss all the shots you don't take.",
        "superlative": "Most Likely to Win the PGA",
      },
      Object {
        "id": 2,
        "name": "Amy",
        "photo": "./amy.jpg",
        "quote": "I believe in you, cuties!",
        "superlative": "Most Likely to Be an Actual Superhero",
      },
    ]
  }
  title="Staff"
>
  <div>
    <h2>
      Staff
    </h2>
    <div
      className="card-area"
    >
      <Person
        id={1}
        key="1"
        name="Eric"
        photo="./eric.jpg"
        quote="You miss all the shots you don't take."
        superlative="Most Likely to Win the PGA"
      >
        <div
          className="person-card"
          key="1"
        >
          <img
            alt="portrait"
            src="./eric.jpg"
          />
          <h3
            className="name"
          >
            Eric
          </h3>
          <p
            className="quote"
          >
            You miss all the shots you don't take.
          </p>
          <p
            className="superlative"
          >
            Most Likely to Win the PGA
          </p>
        </div>
      </Person>
      <Person
        id={2}
        key="2"
        name="Amy"
        photo="./amy.jpg"
        quote="I believe in you, cuties!"
        superlative="Most Likely to Be an Actual Superhero"
      >
        <div
          className="person-card"
          key="2"
        >
          <img
            alt="portrait"
            src="./amy.jpg"
          />
          <h3
            className="name"
          >
            Amy
          </h3>
          <p
            className="quote"
          >
            I believe in you, cuties!
          </p>
          <p
            className="superlative"
          >
            Most Likely to Be an Actual Superhero
          </p>
        </div>
      </Person>
    </div>
  </div>
</Cohort>
`;
